ai/core/src/unitycatalog/ai/core/utils/validation_utils.py

```diff
@@ -158,42 +154,54 @@ def has_retriever_signature(function_info: "FunctionInfo") -> bool:
     Returns:
         bool: If the provided function has a valid retriever signature.
     """
-    if "TABLE_TYPE" not in str(function_info.data_type):
+    valid_columns_dict = {"page_content": "STRING", "id": "STRING", "metadata": "MAP"}
+
+    if "TABLE_TYPE" in str(function_info.data_type):
+        if function_info.return_params and function_info.return_params.parameters:
+            params = []
+            for param in function_info.return_params.parameters:
+                if hasattr(param, "as_dict"):
+                    params.append(param.as_dict())
+                else:
+                    params.append(dict(param))
+
+            param_dict = {param["name"]: param["type_name"] for param in params}
+            return param_dict.items() <= valid_columns_dict.items() and "page_content" in param_dict
```

Comment:
Ok to change this later, but I think it's ok if the UDF includes extra columns besides the ones in MLflow's retriever schema, it should be future-proof to be permissive